src/main/java/org/isf/medicalinventory/gui/InventoryEdit.java

```diff
@@ -1061,14 +1113,23 @@ private JTable getJTableInventoryRow() throws OHServiceException {
 				@Override
 				public void valueChanged(ListSelectionEvent e) {
 					if (e.getValueIsAdjusting()) {
-						jTableInventoryRow.editCellAt(jTableInventoryRow.getSelectedRow(), jTableInventoryRow.getSelectedColumn());
-						int[] selectedRows = jTableInventoryRow.getSelectedRows();
-						if (selectedRows.length == 1) {
-							MedicalInventoryRow medInvRow = (MedicalInventoryRow) jTableInventoryRow.getValueAt(selectedRows[0], -1);
-							if (medInvRow.getLot() == null || medInvRow.isNewLot()) {
+						int selectedRow = jTableInventoryRow.getSelectedRow();
+						if (selectedRow != -1) {
+							MedicalInventoryRow medInvRow = (MedicalInventoryRow) jTableInventoryRow.getValueAt(selectedRow, -1);
+							Lot lot = medInvRow.getLot() != null ? medInvRow.getLot() : null;
+							if (lot == null || medInvRow.isNewLot()) {
 								lotButton.setEnabled(true);
 							} else {
-								lotButton.setEnabled(false);
+								BigDecimal cost = lot.getCost() != null ? lot.getCost() : new BigDecimal(0.00);
+								if (lot != null) {
```

Comment:
could be there cases like `!medInvRow.isNewLot() && lot == null`?

Reply:
No

Reply:
what I mean is that if you already tested `lot == null` and you are in the `else` branch, why to test again if `lot != null`?
please revise this logic because it happens that Lot button is disabled also for new lots now (N)
![image](https://github.com/user-attachments/assets/c280e4bf-13f4-4c00-ade7-8212198cf827)

Reply:
fixe